### DISCUSSION TITLE
nested component states

### DISCUSSION THREAD

**Discussion by hellcoderz** [brendonmatos/golive#27] [Q&A]
I understand we can create nested components but I am not sure how to pass states downstream and how the updates will send upstream for the component state

**Comment by Hugome** [comment]
From my understanding (Very new to golive) you can pass states downstream via `CompoentContext` in `LiveComponent`, passed to childs here :
https://github.com/brendonmatos/golive/blob/d475256504d0b9d0ef9d431c3d885c40261c8363/component.go#L140
And upstream I think you can just pass a reference to a common struct/variable/... or use channel (Currently I use rxgo to have observer and auto commit component when state change, to have "kind of" state management with store like in vue/react/svelte/...)
